src/main/java/seedu/address/model/task/UniqueTaskList.java

```diff
@@ -110,6 +119,30 @@ public void setTasks(List<Task> tasks) {
 
         internalList.setAll(tasks);
     }
+    //==============Check overdue state of tasks =============================
+    /**
+     * Checks overdue state of task and updates all relevant tasks
+     */
+    public void updateOverdue() {
+        for (int i = 0; i < internalList.size(); i++) {
+            Task task = internalList.get(i);
+            if (task.isOverdue()) {
```

Comment:
I think this condition should be `task.isOverdue() && !task.isStatusComplete()`

Reply:
Right. I have updated.